code/modules/food_and_drinks/kitchen_machinery/food_replicator.dm

```diff
@@ -0,0 +1,561 @@
+	var/list/reagent_tiers = list(
+								"tier0" = list("water", "sodiumchloride", "blackpepper", "ice"),
+								"tier1" = list("nutriment", "protein", "plantmatter", "rice", "sugar"),
+								"tier2" = list("ketchup", "berryjuice", "egg", "cocoa", "soysauce", "capsaicin", "tomatojuice", "sprinkles", "cheese", "banana", "toxin",
+												"mashedpotatoes", "poisonberryjuice", "chocolate", "cream", "milk", "soymilk", "lemonjuice", "limejuice", "grapejuice",
+												"orangejuice", "cherryjelly", "watermelonjuice", "vanilla"),
+								"tier3" = list("vitamin", "frostoil", "porktonium", "msg", "fake_cheese", "slimejelly", "coffee", "gravy", "salglu_solution", "amanitin",
+												"psilocybin", "oculine", "blood", "cholestrol", "minttoxin", "space_drugs", "cola", "dr_gibb", "beans",
+												"plasma", "ether", "morphine", "synaptizine", "mannitol"),
+								"tier4" = list("omnizine", "weak_omnizine", "sulfonal")
+								)
+
+	//memory_slots is the number of memory slots you can save dishes to for replication. Upgrading the scanning module improves this, allowing for more stored dish options.
+	var/memory_slots = 1
+	var/list/memory	= null		//this gets built in New(), so we can do with with a loop to cut down on what looks like copy-pasta in the defines
+	var/obj/item/weapon/reagent_containers/food/snacks/loaded = null
+
+/obj/machinery/food_replicator/New()
+	..()
+	component_parts = list()
+	component_parts += new /obj/item/weapon/circuitboard/food_replicator(null)
+	component_parts += new /obj/item/weapon/stock_parts/micro_laser(null)
+	component_parts += new /obj/item/weapon/stock_parts/manipulator(null)
+	component_parts += new /obj/item/weapon/stock_parts/scanning_module(null)
+	component_parts += new /obj/item/weapon/stock_parts/matter_bin(null)
+	component_parts += new /obj/item/weapon/stock_parts/console_screen(null)
+	component_parts += new /obj/item/stack/cable_coil(null, 5)
+	RefreshParts()
+	if(!memory)
+		memory = list()
+		for(var/i in 1 to 4)
+			//memory list breakdown:
+				//"name" is the dish name, used to ensure replicated dishes copy even custom names
+				//"desc" is the dish desc, used to ensure replicated dishes examine the same
+				//"icon" is the dish icon file, used to ensure we can access the proper sprite
+				//"state" is the dish icon_state, used to ensure we look like what we copied
+				//"slice_path" is the dish's sliceable path, if one exists for the original food. This is so you can cut replicated cakes.
+				//"slice_num" is the dish's slices_num, used to ensure we can slice into the proper number of pieces
+				//"reagents" is the complete reagent_list of the dish when saved. This may include reagents we can't replicate at the time.
+			memory.Add("slot[i]" = list("name" = null, "desc" = null, "icon" = null, "state" = null, "slice_path" = null, "slice_num" = null, "reagents" = list()))
```

Comment:
Perhaps, make a proc that returns an associative list, that has each of these attributes as individual arguments